### PR TITLE
Handle server errors the way that Ember Data does

### LINKED ISSUE #22: normalizeErrorResponse should handle JSON responses
My API is returning a JSON response along with an error code status (not any of the ones that are handled), and `normalizeErrorResponse` is converting the payload to a string [here](https://github.com/ember-cli/ember-ajax/blob/master/addon/services/ajax.js#L315). The results in the `detail` of the error object being "[object Object]".

Is this the intended behavior?

### PR DESCRIPTION
**Pull request by alexlafroscia**
Closes #22

### CODE REVIEW

**Comment by taras** [comment]
I'm not seeing any tests that show detailedMessage being exposed. Was it missed or did I misunderstand how Ember Data code works?

**Comment by alexlafroscia** [comment]
It's now the `message` attribute of the error, which seems to be what Ember Data does, right? And I included a test in the PR that verifies that the message is correct.

**Comment by alexlafroscia** [comment]
Is the attribute supposed to be `detailedMessage`? I just passed the new description as the second parameter to the constructor.

**Comment by taras** [comment]
:+1: LTGM great job!
